From c1a68db027dc2d06181d4e4a8a0ba6a0d6f4166a Mon Sep 17 00:00:00 2001
From: Volodymyr Babchuk <volodymyr_babchuk@epam.com>
Date: Fri, 26 Jul 2024 19:53:06 +0300
Subject: [PATCH 20/20] bcm2712: enable linux kernel image header

This change is needed to make OP-TEE operable on RPI5.

This is somewhat convoluted, but in the essence, OP-TEE image is being
loaded at address 0x80000 and we can't change that. Problem is that
last versions (like 2024/06/05) of RPI5 firmware tries to load
non-kernel images to the same address 0x80000, thus, overwriting
OP-TEE image.

This patch enables Linux kernel image for U-Boot, so firmware will
treat U-Boot as a kernel image, placing it at different address,
namely at 0x200000.

Upstream-Status: Pending

Signed-off-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>
Reviewed-by: Oleksii Moisieiev <oleksii_moisieiev@epam.com>
Tested-by: Oleksii Moisieiev <oleksii_moisieiev@epam.com>
Signed-off-by: Xiaolei Wang <xiaolei.wang@windriver.com>
---
 arch/arm/mach-bcm283x/Kconfig | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/arch/arm/mach-bcm283x/Kconfig b/arch/arm/mach-bcm283x/Kconfig
index e7e47b11..3008c306 100644
--- a/arch/arm/mach-bcm283x/Kconfig
+++ b/arch/arm/mach-bcm283x/Kconfig
@@ -58,8 +58,12 @@ config BCM2711_64B
 
 config BCM2712
 	bool "Broadcom BCM2712 SoC support"
+	select LINUX_KERNEL_IMAGE_HEADER
 	depends on ARCH_BCM283X
 
+config LNX_KRNL_IMG_TEXT_OFFSET_BASE
+	default TEXT_BASE
+
 menu "Broadcom BCM283X family"
 	depends on ARCH_BCM283X
 choice
-- 
2.43.0

